addons/account/models/account_payment.py

```diff
@@ -4,15 +4,74 @@
 from odoo.exceptions import UserError, ValidationError
 
 
+class AccountPaymentMethodType(models.Model):
+    _name = "account.payment.method.type"
+    _description = "Method Types"
+
+    name = fields.Char(translate=True, required=True)
+    code = fields.Char(required=True)  # For internal identification
+    category = fields.Selection(selection=[('electronic', 'Electronic'), ('manual', 'Manual')],
+                                required=True)
+    payment_type = fields.Selection(selection=[('inbound', 'Inbound'), ('outbound', 'Outbound')],
+                                    required=True)
+    journal_type = fields.Selection(selection=[('bank', 'Bank'), ('cash', 'Cash')],
+                                    help='Choose on which journal type this method will be usable. '
+                                         'Leave empty to allow both types.',
+                                    required=False)
+
+    _sql_constraints = [
+        ('name_code_unique', 'unique (code, payment_type)', 'The combination code/payment type already exists!'),
+    ]
+
+
 class AccountPaymentMethod(models.Model):
     _name = "account.payment.method"
     _description = "Payment Methods"
-    _order = 'sequence'
 
-    name = fields.Char(required=True, translate=True)
-    code = fields.Char(required=True)  # For internal identification
-    payment_type = fields.Selection([('inbound', 'Inbound'), ('outbound', 'Outbound')], required=True)
-    sequence = fields.Integer(help='Used to order Methods in the form view', default=10)
+    name = fields.Char(compute='_compute_name', readonly=False, store=True)
+    active = fields.Boolean(default=True)
+    sequence = fields.Integer()
+    category = fields.Selection(
+        selection=[('electronic', 'Electronic'), ('manual', 'Manual')],
```

Comment:
BY default, there is no `electronic` right? This should be added by the `payment` module I guess